### PR TITLE
Cron jobs and WSAPI

### PR DESCRIPTION
**Pull request by j-jasik**
2 big things were accomplished in this branch:
1) All cron jobs were implemented
2) The direct connection to banner was replaced by WSAPI

### CODE REVIEW

File: sciencelabs/sessions/__init__.py [hunk 1]
```diff
@@ -399,3 +399,12 @@ def tutor_sign_in(self, session_id):
     def tutor_sign_out(self, session_id, tutor_id):
         self.session.tutor_sign_out(session_id, tutor_id)
         return redirect(url_for('SessionView:tutor_attendance', session_id=session_id))
+
+    @requires_auth
+    @route('/cron_close_sessions', methods=['get'])
+    def cron_close_sessions(self):
+        try:
+            message = self.session.close_open_sessions_cron()
+            return message
```
Comment (c-schwarze) [inline sciencelabs/sessions/__init__.py]: maybe just combine 407/408

File: sciencelabs/users/__init__.py [hunk 2]
```diff
@@ -130,10 +134,28 @@ def create_user(self):
             last_name = form.get('last-name')
             username = form.get('username')
             roles = form.getlist('roles')
-            self.user.create_user(first_name, last_name, username)
+            self.user.create_user(first_name, last_name, username, 0)
```
Comment (c-schwarze) [inline sciencelabs/users/__init__.py]: are you defaulting email pref to 0? I think it should be defaulted to 1. thoughts?

File: sciencelabs/sessions/__init__.py (hunk 1 above)
Comment (j-jasik) [reply sciencelabs/sessions/__init__.py]: Done

File: sciencelabs/users/__init__.py (hunk 2 above)
Comment (j-jasik) [reply sciencelabs/users/__init__.py]: My thought was that the majority of users that get created are students, who we do not want to send emails to. Honestly though how email pref gets set should probably be reliant on the user's roles. Profs and administrators should probably get defaulted to 1, while tutors and students to 0? What do you think? On the current site, students don't even have the option to change their email preferences. They are automatically set to 0 and cannot change that themselves.

Comment (c-schwarze) [reply sciencelabs/users/__init__.py]: Ah yeah you're right. Its a prof/admin specific role 👍

**Pull request merged**